packages/editor/src/components/header/index.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useViewportMatch } from '@wordpress/compose';
+import { __unstableMotion as motion } from '@wordpress/components';
+import { store as preferencesStore } from '@wordpress/preferences';
+import { useState } from '@wordpress/element';
+import { PinnedItems } from '@wordpress/interface';
+
+/**
+ * Internal dependencies
+ */
+import CollapsableBlockToolbar from '../collapsible-block-toolbar';
+import DocumentBar from '../document-bar';
+import DocumentTools from '../document-tools';
+import PostPreviewButton from '../post-preview-button';
+import PostSavedState from '../post-saved-state';
+import PostViewLink from '../post-view-link';
+import PreviewDropdown from '../preview-dropdown';
+import MoreMenu from '../more-menu';
+import PostPublishButtonOrToggle from '../post-publish-button/post-publish-button-or-toggle';
+
+import { store as editorStore } from '../../store';
+
+const slideY = {
+	hidden: { y: '-50px' },
+	distractionFreeInactive: { y: 0 },
+	hover: { y: 0, transition: { type: 'tween', delay: 0.2 } },
+};
+
+function Header( {
+	customSaveButton,
+	forceIsDirty,
+	setEntitiesSavedStatesCallback,
+	children,
+} ) {
+	const isWideViewport = useViewportMatch( 'large' );
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const {
+		isTextEditor,
+		isPublishSidebarOpened,
+		showIconLabels,
+		hasFixedToolbar,
+	} = useSelect( ( select ) => {
+		const { get: getPreference } = select( preferencesStore );
+		const { getEditorMode } = select( editorStore );
+
+		return {
+			isTextEditor: getEditorMode() === 'text',
+			isPublishSidebarOpened:
+				select( editorStore ).isPublishSidebarOpened(),
```

Comment:
Nit: We're already extracting `core/editor` selectors abov; let'ss include `isPublishSidebarOpened`.